src/app/modules/consultations/consultation-profile/consultation-questionnaire/consultation-questionnaire.component.ts

```diff
@@ -0,0 +1,331 @@
+          responseAnswers.push({
+            question_id: item,
+            is_other: true,
+            other_option_answer: answers['other_answer-' + item],
+          });
+        } else if (!(item.includes('other')) && !Array.isArray(answers[item])) {
+          if (answers[item].length > 0 || typeof answers[item] !== 'string') {
+            responseAnswers.push({
+              question_id: item,
+              answer: answers[item]
+            });
+          }
+        }
+      }
+    }
+    return responseAnswers;
+  }
+
+  validateAnswers() {
+    this.consultationService.validateAnswers.subscribe((value) => {
+      if (value) {
+        this.submitAnswer();
+        this.consultationService.validateAnswers.next(false);
+      }
+    });
+  }
+
+  getConsultationResponse() {
+    const consultationResponse =  {
+      consultationId: this.profileData.id,
+      satisfactionRating : this.responseFeedback,
+      visibility: this.responseVisibility ? 'shared' : 'anonymous',
+    };
+    if (this.checkProperties(consultationResponse)) {
+      consultationResponse['templateId'] = this.templateId;
+      consultationResponse['answers'] = this.responseAnswers;
+      return consultationResponse;
+    }
+    return;
+  }
+
+  checkProperties(obj) {
+    for (const key in obj) {
+      if (obj[key] === null ||  obj[key] === '' || obj[key] === undefined) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  onAnswerChange(question?, value?, checkboxValue?) {
+    if (question && value.id === 'other') {
+      let otherValue = true;
+      if (question.questionType === 'checkbox' && value.id === 'other' && !checkboxValue) {
+         otherValue = false;
+      }
+      for (let i = 0; i < this.profileData.questions.length; i++) {
+        if (this.profileData.questions[i].id === question.id) {
+          this.profileData.questions[i].is_other = otherValue;
+          if (question.isOptional) {
+            this.questionnaireForm.addControl('other_answer-' + question.id, new FormControl(null));
+          } else {
+            this.questionnaireForm.addControl('other_answer-' + question.id, new FormControl(null, Validators.required));
+          }
+          break;
+        }
+      }
+    } else {
+      if (question.questionType !== 'checkbox') {
+        this.profileData.questions.forEach(ques => {
+          if (question.id === ques.id) {
+            ques.is_other = false;
+          }
+        });
+      }
+    }
+  }
+
+  subscribeUseTheResponseAnswer() {
+    this.consultationService.useThisResponseAnswer.subscribe((obj: any) => {
+      if (obj && !isObjectEmpty(obj)) {
+        const {templateId, longTextResponses} = obj;
+        this.templateId = templateId;
+        longTextResponses.map((response) => {
+          const controlName = response.id.toString();
+          this.responseText = this.templateText = response.answer;
+          const checkTextAreaElementExist = setInterval(() => {
```

Comment:
@avikt added setInterval since it executed before templated loaded and returning textarea element not exist, so checking till component loaded by set interval

Reply:
@Nishanthh Hmm then would having subscribeUseTheResponseAnswer() in ngAfterViewInit() solve this problem?

Reply:
@avikt yeah solved the problem bro.